### ISSUE TITLE
Accessibility - Focus not visible when menu is opened (SC 2.4.7)

### ISSUE THREAD

**Issue by LaurensBurger** [open-formulieren/open-forms#5459]
### Thema / Theme

Frontend

### Omschrijving / Description

The "openforms-progress-indicator__form-title" does not show a keyboard focus then the menu is open:

<img width="1144" height="635" alt="Image" src="https://github.com/user-attachments/assets/3b011a2c-2102-43be-a3f1-b374ba8c8dd3" />

it should show the focus like the links in the menu:

<img width="1144" height="635" alt="Image" src="https://github.com/user-attachments/assets/6d94c262-0048-4f42-89dd-a5fdf623f58e" />
